pkg/services/blockchain/eth/resolver.go

```diff
@@ -148,9 +241,200 @@ func (r *Resolver) Resolve(
 		}
 	}
 
+	signature := ""
+	if r.walletKey != "" && opts.Signature != "" {
+		primaryType := services.IdentityStateType
+		if opts.GistRoot != nil {
+			primaryType = services.GlobalStateType
+		}
+		signature, err = r.signTypedData(primaryType, did, identityState)
+		if err != nil {
+			return services.IdentityState{}, err
+		}
+	}
+
+	identityState.Signature = signature
+
 	return identityState, err
 }
 
+func (r *Resolver) VerifyState(
+	primaryType services.PrimaryType,
+	identityState services.IdentityState,
+	did w3c.DID,
+) (bool, error) {
+	walletAddress, err := r.WalletAddress()
+	if err != nil {
+		return false, err
+	}
+
+	typedData, err := r.TypedData(primaryType, did, identityState, walletAddress)
+	if err != nil {
+		return false, err
+	}
+
+	authData := AuthData{TypedData: typedData, Signature: identityState.Signature, Address: walletAddress}
+	return r.verifyTypedData(authData)
+}
+
+func TimeStampFn() string {
+	timestamp := strconv.FormatInt(time.Now().UTC().Unix(), 10)
+	return timestamp
+}
+
+func (r *Resolver) TypedData(primaryType services.PrimaryType, did w3c.DID, identityState services.IdentityState, walletAddress string) (apitypes.TypedData, error) {
```

Comment:
We only return 1 message in eip712. GlobalState or IdentityState for signing because in the case of the user we only need GlobalState and in the case of the issuer we only need IdentityState.
Fixed information inside each case.